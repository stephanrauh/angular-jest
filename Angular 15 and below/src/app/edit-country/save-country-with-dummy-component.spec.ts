import { HttpClientTestingModule } from "@angular/common/http/testing";
import { Component } from "@angular/core";
import { ComponentFixture, TestBed } from "@angular/core/testing";
import { NoopAnimationsModule } from "@angular/platform-browser/animations";
import { RouterTestingModule } from "@angular/router/testing";
import { ConfirmationService } from "primeng/api";
import { ConfirmDialogModule } from "primeng/confirmdialog";

@Component({template: `<p-confirmDialog
   key="confirm-save-country"
   message="Do you want to save the country?">
</p-confirmDialog>`})
class DummyComponent {}

describe('Save-Country-Question', () => {
  let fixture: ComponentFixture<DummyComponent>;

  let component: DummyComponent;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [DummyComponent],
      providers: [ConfirmationService],
      imports: [ConfirmDialogModule, RouterTestingModule, HttpClientTestingModule, NoopAnimationsModule],
    }).compileComponents();

    fixture = TestBed.createComponent(DummyComponent);
    component = fixture.componentInstance;

    fixture.detectChanges();
  });

  it("Should render the confirmation dialog", () => {
    const service = TestBed.inject(ConfirmationService);;
    service.confirm({
      key: 'confirm-save-country',
    });
    fixture.detectChanges();
    const msgSpan = document.querySelector(".p-confirm-dialog-message") as HTMLElement;
    expect(msgSpan.innerHTML).toBe("Do you want to save the country?");
    expect(document.body).toMatchSnapshot();
  })
});
