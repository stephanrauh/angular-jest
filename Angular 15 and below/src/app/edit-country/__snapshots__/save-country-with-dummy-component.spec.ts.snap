// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Save-Country-Question Should render the confirmation dialog 1`] = `
<body>
  <div
    id="root0"
  >
    <p-confirmdialog
      class="p-element"
      key="confirm-save-country"
      message="Do you want to save the country?"
    >
      <div
        class="p-dialog-mask p-component-overlay p-dialog-mask-scrollblocker"
      >
        <div
          class="ng-trigger ng-trigger-animation p-dialog p-confirm-dialog p-component ng-animating"
          style=""
        >
          
          <div
            class="p-dialog-header"
            style=""
          >
            
            <div
              class="p-dialog-header-icons"
            >
              <button
                class="p-dialog-header-icon p-dialog-header-close p-link"
                type="button"
              >
                <span
                  class="pi pi-times"
                />
              </button>
              
            </div>
          </div>
          
          <div
            class="p-dialog-content"
          >
            
            <span
              class="p-confirm-dialog-message"
            >
              Do you want to save the country?
            </span>
          </div>
          
          <div
            class="p-dialog-footer"
            style=""
          >
            <button
              class="p-element p-ripple p-confirm-dialog-reject p-button p-component"
              pbutton=""
              pripple=""
              type="button"
            >
              <span
                aria-hidden="true"
                class="p-button-icon p-button-icon-left pi pi-times"
              />
              <span
                class="p-button-label"
              >
                No
              </span>
            </button>
            
            <button
              class="p-element p-ripple p-confirm-dialog-accept p-button p-component"
              pbutton=""
              pripple=""
              type="button"
            >
              <span
                aria-hidden="true"
                class="p-button-icon p-button-icon-left pi pi-check"
              />
              <span
                class="p-button-label"
              >
                Yes
              </span>
            </button>
            
          </div>
          
        </div>
        
      </div>
      
    </p-confirmdialog>
  </div>
</body>
`;
